src/phase0/state_transition/epoch_processing.rs

```diff
@@ -315,8 +315,18 @@ pub fn process_historical_roots_update<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    let next_epoch = get_current_epoch(state, context) + 1;
+    if (next_epoch % (SLOTS_PER_HISTORICAL_ROOT as u64 / context.slots_per_epoch)) == 0 {
```

Comment:
this is not very clear and i even started out using the const values like you did to cut down on needing to pass around a `Context` and hopefully move some computation from runtime to compile-time.

however, i think at the moment for consistency/uniformity i'm keeping the const params at the type-level and where needed referring to values from the `context`

so then this would look like:

```suggestion
    let epochs_per_historical_root = context.slots_per_historical_root / context.slots_per_epoch;
    if next_epoch % epochs_per_historical_root == 0 {
```

where i pulled out the boundary computation for readability (and dropped what i think are extra parens) and also implied a change in the `Context` where the `slots_per_historical_root` field has type `Slot` (it has type `usize` now)

so also this change:

```rust
    pub slots_per_historical_root: Slot,
```

i don't think this type change will break anything as the other constraint is needed `usize` at the type-level but that is handled by the const generic param.